src/devices/Mcp25xxx/Mcp25xxx.cs

```diff
@@ -0,0 +1,379 @@
+        /// <param name="rx1bf">The input pin number that is connected to Rx1BF.</param>
+        /// <param name="clkout">The input pin number that is connected to CLKOUT.</param>
+        /// <param name="masterGpioController">
+        /// The GPIO controller for defined external pins. If not specified, the default controller will be used.
+        /// </param>
+        public Mcp25xxx(
+            SpiDevice spiDevice,
+            int reset = -1,
+            int tx0rts = -1,
+            int tx1rts = -1,
+            int tx2rts = -1,
+            int interrupt = -1,
+            int rx0bf = -1,
+            int rx1bf = -1,
+            int clkout = -1,
+            IGpioController masterGpioController = null)
+        {
+            _spiDevice = spiDevice;
+
+            _reset = reset;
+            _tx0rts = tx0rts;
+            _tx1rts = tx1rts;
+            _tx2rts = tx2rts;
+            _interrupt = interrupt;
+            _rx0bf = rx0bf;
+            _rx1bf = rx1bf;
+            _clkout = clkout;
+
+            // Only need master controller if there are external pins provided.
+            if (_reset != -1 ||
+                _tx0rts != -1 ||
+                _tx1rts != -1 ||
+                _tx2rts != -1 ||
+                _interrupt != -1 ||
+                _rx0bf != -1 ||
+                _rx1bf != -1 ||
+                _clkout != -1)
+            {
+                _masterGpioController = masterGpioController ?? new GpioController();
+
+                if (_tx0rts != -1)
+                {
+                    _masterGpioController.OpenPin(_tx0rts, PinMode.Output);
+                }
+
+                if (_tx1rts != -1)
+                {
+                    _masterGpioController.OpenPin(_tx1rts, PinMode.Output);
+                }
+
+                if (_tx2rts != -1)
+                {
+                    _masterGpioController.OpenPin(_tx2rts, PinMode.Output);
+                }
+
+                if (_interrupt != -1)
+                {
+                    _masterGpioController.OpenPin(_interrupt, PinMode.Input);
+                }
+
+                if (_rx0bf != -1)
+                {
+                    _masterGpioController.OpenPin(_rx0bf, PinMode.Input);
+                }
+
+                if (_rx1bf != -1)
+                {
+                    _masterGpioController.OpenPin(_rx1bf, PinMode.Input);
+                }
+
+                if (_clkout != -1)
+                {
+                    _masterGpioController.OpenPin(_clkout, PinMode.Input);
+                }
+
+                if (_reset != -1)
+                {
+                    _masterGpioController.OpenPin(_reset, PinMode.Output);
+                    WriteResetPin(PinValue.Low);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes a value to Tx0RTS pin.
+        /// </summary>
+        public void WriteTx0RtsPin(PinValue value)
```

Comment:
I'm wondering if this should just be:
```csharp
PinValue Tx0Rts { get { ... } set { ... } }
```

Reply:
Yeah, I wasn't liking this naming either.  There are a mixture of pins that are inputs or outputs.  So in this case we could do this...

```csharp
// Add common helper for checking.
private void VerifyPinConfigured(int pin, string name)
{
    if (pin == -1)
    {
        throw new InvalidOperationException($"No {name} pin configured.");
    }
}

// For output pins...
public PinValue Tx0Rts
{
    set
    {
        VerifyPinConfigured(_tx0rts, nameof(Tx0Rts));
        _masterGpioController.Write(_tx0rts, value);
    }
}

// For input pins...
public PinValue Rx0Bf
{
    get
    {
        VerifyPinConfigured(_rx0bf, nameof(Rx0Bf));
        return _masterGpioController.Read(_rx0bf);
    }
}
```